src/filesystem/drive/access.rs

```diff
@@ -51,6 +51,19 @@ impl DriveAccess {
             .clone()
     }
 
+    pub(crate) fn verifying_keys(&self) -> Vec<VerifyingKey> {
+        self.actor_settings
+            .iter()
+            .filter_map(|(actor_id, settings)| {
+                if self.is_owner(actor_id) {
+                    Some(settings.verifying_key())
+                } else {
+                    None
+                }
+            })
+            .collect()
+    }
+
```

Comment:
We should definitely not be doing any kind of filtering on keys at this level. It should be up to the caller on the Drive's exposed function that should perform any filtering. At this level we don't know what the intent of the caller is but generally its almost always going to be more than just the owner keys (even in this case we want more than just the owner keys).